forte/data/data_store.py

```diff
@@ -480,7 +501,13 @@ def next_entry(self, tid: int) -> List:
         Returns:
             The next entry of the same type as the `tid` entry.
         """
-        raise NotImplementedError
+        _, type_id, index_id = self.get_entry(tid=tid)
+        entry_list = self.__elements[type_id]
+        if index_id >= len(entry_list) - 1:
+            raise IndexError(
+                f"Entry with tid {tid} is the last entry in entry list."
+            )
+        return entry_list[index_id + 1]
 
     def prev_entry(self, tid: int) -> List:
```

Comment:
It should be important to document the list ordering of the non-annotation types. What would happen if we query `prev_entry` or `next_entry` on those? @wanglec

Reply:
if it is a non-annotation type, it will be sorted in the insertion order, which means the next and prev would return the next/previous inserted entry.

Reply:
We also change the structure of `self.__elements` in data_pack.py. Sorry for the confusion in the string description.
it looks like this:
{"ft.onto.base_ontology.Document": SortedList([]), "ft.onto.base_ontology.Sentence":  SortedList([]) }, etc.

Reply:
Thanks for the update! I've fixed the implementation based on #665.

Reply:
> if it is a non-annotation type, it will be sorted in the insertion order, which means the next and prev would return the next/previous inserted entry.

Let's clarify these behaviors in docstring